aws/resource_aws_elasticsearch_domain.go

```diff
@@ -644,34 +641,30 @@ func resourceAwsElasticSearchDomainRead(d *schema.ResourceData, meta interface{}
 
 	d.Set("arn", ds.ARN)
 
-	listOut, err := conn.ListTags(&elasticsearch.ListTagsInput{
-		ARN: ds.ARN,
-	})
+	tags, err := keyvaluetags.ElasticsearchserviceListTags(conn, d.Id())
 
 	if err != nil {
-		return err
-	}
-	var est []*elasticsearch.Tag
-	if len(listOut.TagList) > 0 {
-		est = listOut.TagList
+		return fmt.Errorf("error listing tags for Elasticsearch Cluster (%s): %s", d.Id(), err)
 	}
 
-	d.Set("tags", tagsToMapElasticsearchService(est))
+	if err := d.Set("tags", tags.IgnoreAws().Map()); err != nil {
+		return fmt.Errorf("error setting tags: %s", err)
```

Comment:
`d.Set()` error handling tends to be in the (undocumented/unlinted) form:

```go
if err := d.Set("ATTR", /* ... */); err != nil {
  return fmt.Errorf("error setting ATTR: %s", err)
}
```

See also line 601 above.

When the [Terraform Plugin SDK supports typed `d.Get()`/`d.Set()` functions](https://github.com/hashicorp/terraform-plugin-sdk/issues/229), we can hopefully start removing some of this boilerplate error handling for maps since the compiler could catch any typing issues, e.g.

```go
d.SetMap("tags", tags.IgnoreAws().Map())
```